### PR TITLE
Authpolicy v1beta2 reformat

### PR DESCRIPTION
**Pull request by azgabur**
This PR makes this testsuite compatible with newest AuthConfig/AuthPolicy v1beta2

Few notes:
- Envoy Dynamic Metadata function needs to be added in Responses section when tests for it will be created
    - **Wristband tests are for now skipped!** 
- Test for Custom deny code/body/headers need to be created
- testsuite/tests/kuadrant/authorino/response/test_headers.py is maybe not needed as wrapperKey no longer exists or needs some rewrite

### CODE REVIEW

File: testsuite/openshift/objects/auth_config/sections.py [hunk 1]
```diff
@@ -208,7 +230,9 @@ def add_simple(self, auth_json: str, name="simple", key="data", **common_feature
     @modify
     def add_json(self, name: str, properties: list[Property], **common_features):
         """Adds json response to AuthConfig"""
-        asdict_properties = [asdict(p) for p in properties]
+        asdict_properties = {}
+        for prop in properties:
+            asdict_properties.update(asdict(prop))
```
Comment (pehala) [inline testsuite/openshift/objects/auth_config/sections.py:235]: I do not follow this change, does this mean properties can override each other?

Comment (averevki) [reply testsuite/openshift/objects/auth_config/sections.py:235]: I think it's just a way to populate a dict with other dicts in the loop.

*You probably can use something like this to disable overwriting if you want.
```suggestion
            asdict_properties = {**prop, **asdict_properties}
```

Comment (pehala) [reply testsuite/openshift/objects/auth_config/sections.py:235]: I meant that before it was a list, meaning all properties were separated, but now it is a one-level dict so that they can override each other. That change is fundamental on the AuthPolicy/AuthConfig level.

Previously we used List because in the CR it was a list, now it is changed to a dict, but we kept the list, which can cause confusion. IF I specify two properties with the same name, I would expect both of them to be propagated to the CR, but now they wont, only one will be.

Basically, I think maybe we should change to dict as well to keep it consistent.

Comment (averevki) [reply testsuite/openshift/objects/auth_config/sections.py:235]: Makes sense about possible confusion. Although, the behavior should be the same. @azgabur what would happen if you specify 2 properties with the same name? Would it be overridden / raise an error during creation?

File: testsuite/tests/kuadrant/authorino/response/test_headers.py [hunk 2]
```diff
@@ -16,11 +16,11 @@ def header_name(request):
 def authorization(authorization, header_name):
     """Add response to Authorization"""
     authorization.responses.clear_all()  # delete previous responses due to the parametrization
-    authorization.responses.add_json("header", [Property("anything", Value("one"))], wrapper_key=header_name)
```
Comment (averevki) [inline testsuite/tests/kuadrant/authorino/response/test_headers.py:19]: `operator.http.conftest` also requires a change with wrapper_key to work properly

File: testsuite/openshift/objects/auth_config/sections.py [hunk 3]
```diff
@@ -180,23 +198,27 @@ def add_uma(self, name, endpoint, credentials_secret, **common_features):
 
 
 class ResponseSection(Section):
-    """Section which contains response configuration"""
+    """Section which contains response configuration. todo envoyDynamicMetadata"""
```
Comment (averevki) [inline testsuite/openshift/objects/auth_config/sections.py:201]: If we are leaving dynamic metadata implementation for the future PR, we should somehow mark wristband tests as they are currently raising an error

File: testsuite/openshift/objects/auth_config/__init__.py [hunk 4]
```diff
@@ -25,7 +25,7 @@ def authorization(self) -> AuthorizationSection:
     @cached_property
     def identity(self) -> IdentitySection:
         """Gives access to identity settings"""
-        return IdentitySection(self, "identity")
+        return IdentitySection(self, "authentication")
```
Comment (averevki) [inline testsuite/openshift/objects/auth_config/__init__.py:28]: should we also rename `identity` tests package to `authentication` to avoid confusion?

**Review by averevki — COMMENTED** [review]
rhsso tests don't work on my machine (I don't know if they were working before 🙂)

File: testsuite/openshift/objects/auth_config/sections.py (hunk 1 above)
Comment (pehala) [reply testsuite/openshift/objects/auth_config/sections.py:235]: Also sidenote: this does not have to solved in this PR specifically, follow-up PR is sufficient

File: testsuite/openshift/objects/auth_config/__init__.py (hunk 4 above)
Comment (pehala) [reply testsuite/openshift/objects/auth_config/__init__.py:28]: I agree, but can be done later

File: testsuite/openshift/objects/auth_config/sections.py (hunk 1 above)
Comment (azgabur) [reply testsuite/openshift/objects/auth_config/sections.py:235]: I think two properties with same name are no longer possible with the new version - will raise OpenShiftError. Same with other places that it was changed from list to dict.  
Changing the function signature to dict would make object `Property` redundant as the name will now be a key of the dict. I can do the change.

File: testsuite/openshift/objects/auth_config/sections.py [hunk 5]
```diff
@@ -42,29 +61,12 @@ def committed(self):
...
-        if when:
-            item["when"] = [asdict(x) for x in when]
-        if metrics:
-            item["metrics"] = metrics
-        if cache:
-            item["cache"] = asdict(cache)
-        if priority:
-            item["priority"] = priority
-        self.section.append(item)
+        add_common_features(value, **common_features)
```
[the remark this answers is not in the document]
Comment (azgabur) [reply testsuite/openshift/objects/auth_config/sections.py:68]: I extracted it to static function as I needed a way to add common features for new Responses add function. I am not sure if its really needed - meaning I am not sure if Responses section uses some common features. I need to research this more.

File: testsuite/openshift/objects/auth_config/sections.py (hunk 3 above)
Comment (azgabur) [reply testsuite/openshift/objects/auth_config/sections.py:200]: Added pytest skip

File: testsuite/tests/kuadrant/authorino/response/test_headers.py (hunk 2 above)
Comment (azgabur) [reply testsuite/tests/kuadrant/authorino/response/test_headers.py:19]: Fixed

File: testsuite/objects/__init__.py [hunk 6]
```diff
@@ -73,12 +73,14 @@ def __post_init__(self):
...
 
-    in_location: str
+    in_location: Literal["authorizationHeader", "customHeader", "queryString", "cookie"]
     keySelector: str
 
     def asdict(self):
-        """Custom asdict, because I cannot use 'in' as a name"""
-        return {"in": self.in_location, "keySelector": self.keySelector}
+        """Custom asdict because of needing to put location as parent dict key for inner dict"""
+        if self.in_location == "authorizationHeader":
```
[the remark this answers is not in the document]
Comment (pehala) [reply testsuite/objects/__init__.py:81]: In this case I do not think Abstract class is required but yes

[the remark this answers is not in the document]
Comment (azgabur) [reply testsuite/objects/__init__.py]: I was thinking about this for some time but cant find any solution that fits better as just custom `asdict` as its in commit https://github.com/Kuadrant/testsuite/pull/248/commits/88fed67d671a04712841c950522b540914ab8408. Please take a look at my new commit https://github.com/Kuadrant/testsuite/pull/248/commits/4673136b28eac5af9b0e3c294397e841fa299eb4 as it is my best idea and if you think its okay I will do tests refactor. Otherwise please advise how would you like this to look.

[the remark this answers is not in the document]
Comment (pehala) [reply testsuite/objects/__init__.py]: I do not think the assert is needed, I would simply remove the value from in_header Literal to indicate what values are accepted in the main class

[the remark this answers is not in the document]
Comment (azgabur) [reply testsuite/objects/__init__.py:81]: I reverted, please make your refactor PR as we discussed. Thanks!

**Review by pehala — DISMISSED** [review]

**Comment by azgabur** [comment]
^ Additional place for renamed Credentials locations I forgot about

File: testsuite/openshift/objects/auth_config/sections.py [hunk 7]
```diff
@@ -180,35 +197,41 @@ def add_uma(self, name, endpoint, credentials_secret, **common_features):
...
     ):
         """Add response to AuthConfig"""
-        if wrapper:
-            value["wrapper"] = wrapper
-        if wrapper_key:
-            value["wrapperKey"] = wrapper_key
-
-        self.add_item(name, value, **common_features)
+        add_common_features(value, **common_features)
+        self.success_headers.update({name: value})
```
Comment (averevki) [inline testsuite/openshift/objects/auth_config/sections.py:220]: are these `success_headers` used somewhere in the testsuite?

**Review by pehala — APPROVED** [review]

**Review by averevki — APPROVED** [review]

File: testsuite/openshift/objects/auth_config/sections.py (hunk 7 above)
Comment (azgabur) [reply testsuite/openshift/objects/auth_config/sections.py:220]: Nope. Just in the `_add` function.
